homeassistant/components/switch/flux.py

```diff
@@ -0,0 +1,227 @@
+    flux = FluxSwitch(name, hass, False, None, lights, start_time, stop_time,
+                      start_colortemp, sunset_colortemp, stop_colortemp,
+                      brightness)
+    add_devices([flux])
+
+    def update(call=None):
+        """Update lights."""
+        flux.flux_update()
+
+    hass.services.register(DOMAIN, 'flux_update', update)
+
+
+# pylint: disable=too-many-instance-attributes
+class FluxSwitch(SwitchDevice):
+    """Flux switch."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, name, hass, state, icon, lights, start_time, stop_time,
+                 start_colortemp, sunset_colortemp, stop_colortemp,
+                 brightness):
+        """Initialize the Flux switch."""
+        self._name = name
+        self.hass = hass
+        self._state = state
+        self._icon = icon
+        self._assumed = False
+        self._lights = lights
+        self._start_time = start_time
+        self._stop_time = stop_time
+        self._start_colortemp = start_colortemp
+        self._sunset_colortemp = sunset_colortemp
+        self._stop_colortemp = stop_colortemp
+        self._brightness = brightness
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo switch."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use for device if any."""
+        return self._icon
+
+    @property
+    def assumed_state(self):
+        """Return if the state is based on assumptions."""
+        return self._assumed
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self._state
+
+    @property
+    def start_time(self):
+        """Return flux start time."""
+        return self._start_time
+
+    @property
+    def stop_time(self):
+        """Return flux stop time."""
+        return self._stop_time
+
+    @property
+    def start_colortemp(self):
+        """Return flux start color temperature."""
+        return self._start_colortemp
+
+    @property
+    def sunset_colortemp(self):
+        """Return flux sunset color temperature."""
+        return self._sunset_colortemp
+
+    @property
+    def stop_colortemp(self):
+        """Return flux stop color temperature."""
+        return self._stop_colortemp
+
+    @property
+    def brightness(self):
+        """Return flux brightness."""
+        return self._brightness
+
+    def turn_on(self, **kwargs):
+        """Turn on flux."""
+        self._state = True
+        self.update_ha_state()
+
+    def turn_off(self, **kwargs):
+        """Turn off flux."""
+        self._state = False
+        self.update_ha_state()
+
+    # pylint: disable=too-many-locals
+    def update(self):
+        """Update the lights if switch is on."""
+        if self._state:
+            self.flux_update()
```

Comment:
I would:

- Remove `update` method
- Calling turn_on will register a listener to call `flux_update` every 30 seconds. (keep setting state and calling update_ha_state)
- Calling turn_off will unregister that listener. (keep setting state and calling update_ha_state)
- Should poll is set to False. (Entity has own timed listener when needed.)

Note that registering a time listener with `track_time_changed_utc` will return the actual listener that got registered. That is the listener that has to be passed in to the remove listener method on `hass.bus`.